File: DemoServer/AdditionalLanguages/python/compareExchange/createCompareExchange/createCompareExchange.py
from demo_example import Example

#region Usings
from ravendb import (
    PutCompareExchangeValueOperation,
    GetCompareExchangeValueOperation
)
#endregion

class RunParams:
    def __init__(self, cmp_xchg_key: str = None, cmp_xchg_value: str = None):
        self.cmp_xchg_key = cmp_xchg_key
        self.cmp_xchg_value = cmp_xchg_value

class CreateCompareExchange(Example):
    def run(self, run_params: RunParams) -> str:
        cmpXchgKey = run_params.cmp_xchg_key or "abc@gmail.com"
        cmpXchgValue = run_params.cmp_xchg_value or "employee/1-A"
        result = None

        #region Demo
        #region Step_1
        put_cmp_xchg_operation = PutCompareExchangeValueOperation(cmpXchgKey, cmpXchgValue, 0)
        put_cmp_xchg_result = self.document_store_holder.store().operations.send(put_cmp_xchg_operation)
        #endregion

        #region Step_2
        success = put_cmp_xchg_result.successful
        put_value = put_cmp_xchg_result.value
        put_version = put_cmp_xchg_result.index

        if not success:
            result = "Key already exists"
        #endregion

        #region Step_3
        get_cmp_xchg_operation = GetCompareExchangeValueOperation(cmpXchgKey, str)
        get_cmp_xchg_result = self.document_store_holder.store().operations.send(get_cmp_xchg_operation)
        #endregion

        #region Step_4
        key = get_cmp_xchg_result.key
        current_value = get_cmp_xchg_result.value
        current_value_version = get_cmp_xchg_result.index
        current_metadata = get_cmp_xchg_result.metadata
        #endregion
        #endregion

        results = (
            result
            or f"Created a new Compare-Exchange "
               f"Key: {key}, "
               f"Value: {current_value}, "
               f"Value Version: {current_value_version}"
        )

        return results

File: DemoServer/AdditionalLanguages/php/compareExchange/createCompareExchange/CreateCompareExchange.php
<?php

namespace RavenDB\Demo\compareExchange\createCompareExchange;

//region Using
use RavenDB\Documents\Operations\CompareExchange\CompareExchangeResult;
use RavenDB\Documents\Operations\CompareExchange\CompareExchangeValue;
use RavenDB\Documents\Operations\CompareExchange\GetCompareExchangeValueOperation;
use RavenDB\Documents\Operations\CompareExchange\PutCompareExchangeValueOperation;
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;

class CreateCompareExchange
{
    public function __invoke(RunParams $runParams): string
    {
        $cmpXchgKey = $runParams->getCmpXchgKey() ?? "abc@gmail.com";
        $cmpXchgValue = $runParams->getCmpXchgValue() ?? "employee/1-A";
        $result = null;

        //region Demo
        //region Step_1
        $putCmpXchgOperation = new PutCompareExchangeValueOperation($cmpXchgKey, $cmpXchgValue, 0);

        /** @var CompareExchangeResult $putCmpXchgResult */
        $putCmpXchgResult = DocumentStoreHolder::getStore()->operations()->send($putCmpXchgOperation);
        //endregion

        //region Step_2
        $success = $putCmpXchgResult->isSuccessful();
        $putValue = $putCmpXchgResult->getValue();
        $putVersion = $putCmpXchgResult->getIndex();

        if (!$success) {
            $result = "Key already exists";
        }
        //endregion

        //region Step_3
        $getCmpXchgOperation = new GetCompareExchangeValueOperation(null, $cmpXchgKey);

        /** @var CompareExchangeValue $getCmpXchgResult */
        $getCmpXchgResult = DocumentStoreHolder::getStore()->operations()->send($getCmpXchgOperation);
        //endregion

        //region Step_4
        $key = $getCmpXchgResult->getKey();
        $currentValue = $getCmpXchgResult->getValue();
        $currentValueVersion = $getCmpXchgResult->getIndex();
        $currentMetadata = $getCmpXchgResult->getMetadata();
        //endregion
        //endregion

        $result = $result ?? "Created a new Compare-Exchange Key: " . $key . ", Value: " . $currentValue . ", Value Version: " . $currentValueVersion;

        return $result;
    }
}
